src/components/LoginForm.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+import { Grid, Button, Form, Input } from "semantic-ui-react";
+import auth from "../modules/auth";
+import { useHistory } from "react-router-dom";
+import "../css/index.css";
+import { Link } from "react-router-dom";
+
+const LoginForm = (props) => {
+  const [message, setMessage] = useState("");
+  const history = useHistory();
+
+  const login = async (e) => {
+    try {
+      const response = await auth.signIn(
+        e.target.email.value,
+        e.target.password.value
+      );
+      if (response.success) {
+        props.setUid(response.data.uid);
+        props.setAuthenticated(true);
+        history.push("/");
```

Comment:
Would it be possible to return the user to where they came from?
Maybe they visit a premium article, they realize they are not logged in, then they log in, and lose track of the article.

Reply:
we will look at that